chaoslib/litmus/node_cpu_hog/node-cpu-hog.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/litmuschaos/litmus-go/pkg/status"
+	"github.com/litmuschaos/litmus-go/pkg/types"
+	"github.com/openebs/maya/pkg/util/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var err error
+
+// PrepareNodeCPUHog contains prepration steps before chaos injection
+func PrepareNodeCPUHog(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets, resultDetails *types.ResultDetails, eventsDetails *types.EventDetails, chaosDetails *types.ChaosDetails) error {
+
+	//Select the node name
+	appNodeName, err := GetNodeName(experimentsDetails, clients)
+	if err != nil {
+		return errors.Errorf("Unable to get the node name due to, err: %v", err)
+	}
+
+	// When number of cpu cores for hogging is not defined , it will take it from node capacity
+	if experimentsDetails.NodeCPUcores == 0 {
+		err = SetCPUCapacity(experimentsDetails, appNodeName, clients)
+	}
+
+	log.InfoWithValues("[Info]: Details of application under chaos injection", logrus.Fields{
+		"NodeName":     appNodeName,
+		"NodeCPUcores": experimentsDetails.NodeCPUcores,
+	})
+
+	experimentsDetails.RunID = GetRunID()
+
+	//Waiting for the ramp time before chaos injection
+	if experimentsDetails.RampTime != 0 {
+		log.Infof("[Ramp]: Waiting for the %vs ramp time before injecting chaos", strconv.Itoa(experimentsDetails.RampTime))
+		waitForRampTime(experimentsDetails)
+	}
+
+	if experimentsDetails.EngineName != "" {
+		msg := "Injecting " + experimentsDetails.ExperimentName + " chaos on " + appNodeName + " node"
+		types.SetEngineEventAttributes(eventsDetails, types.ChaosInject, msg, chaosDetails)
+		events.GenerateEvents(eventsDetails, clients, chaosDetails, "ChaosEngine")
+	}
+
+	// Creating the helper pod to perform node cpu hog
+	err = CreateHelperPod(experimentsDetails, clients, appNodeName)
+	if err != nil {
+		errors.Errorf("Unable to create the helper pod, err: %v", err)
+	}
+
+	//Checking the status of helper pod
+	log.Info("[Status]: Checking the status of the helper pod")
+	err = status.CheckApplicationStatus(experimentsDetails.ChaosNamespace, "name=node-cpu-hog-"+experimentsDetails.RunID, clients)
+	if err != nil {
+		return errors.Errorf("helper pod is not in running state, err: %v", err)
+	}
+
+	// Wait till the completion of helper pod
+	log.Infof("[Wait]: Waiting for %vs till the completion of the helper pod", strconv.Itoa(experimentsDetails.ChaosDuration+30))
+
+	err = status.WaitForCompletion(experimentsDetails.ChaosNamespace, "name=node-cpu-hog-"+experimentsDetails.RunID, clients, experimentsDetails.ChaosDuration+30)
+	if err != nil {
+		return err
+	}
+
+	// Checking the status of application node
+	log.Info("[Status]: Getting the status of application node")
+	err = status.CheckNodeStatus(appNodeName, clients)
+	if err != nil {
+		log.Warn("Application node is not in ready state, Please do a manual recovery for this")
```

Comment:
```suggestion
		log.Warn("Application node is not in the ready state, you may need to manually recover the node")
```

It is possible that managed k8s nodes will self-heal also